per/views.py

```diff
@@ -73,406 +37,300 @@ def handle_post(self, request, *args, **kwargs):
+                    for rubr in data:
+                        FormData.objects.create(form=form,
+                                                question_id=rubr.id,
+                                                selected_option=rubr.op,
+                                                notes=rubr.nt)
+            except Exception:
+                return bad_request('Could not insert PER formdata record.')
 
-        required_fields = [
-            'id',
-        ]
+        return JsonResponse({'status': 'ok'})
 
-        currentDT = datetime.datetime.now(pytz.timezone('UTC'))
-        if 'ended_at' not in body:
-            body['ended_at']     = str(currentDT)
-        if 'submitted_at' not in body:
-            body['submitted_at'] = str(currentDT)
-
-        missing_fields = [field for field in required_fields if field not in body]
-        if len(missing_fields):
-            return bad_request('Could not complete request. Please submit %s' % ', '.join(missing_fields))
 
+class FormEdit(APIView):
+    authentication_classes = (authentication.TokenAuthentication,)
+    permissions_classes = (permissions.IsAuthenticated,)
+
+    def post(self, request):
+        form_id = request.data.get('id', None)
+        if form_id:
+            return bad_request('Could not complete request. Please submit %s' % form_id)
+
+        form = Form.objects.filter(pk=form_id).first()
+        if form is None:
+            return bad_request('Could not find PER form record.')
+
+        started_at = request.data.get('started_at', form.started_at)
+        ended_at = request.data.get('ended_at', get_now_str())
+        submitted_at = request.data.get('submitted_at', get_now_str())
+        name = request.data.get('name', form.name)
+        language = request.data.get('language', form.language)
+        country_id = request.data.get('country_id', form.country_id)
+        ns = request.data.get('ns', form.ns)
+        ip = request.data.get('ip_address', form.ip_address)
+        comment = request.data.get('comment', form.comment)
+        validated = request.data.get('validated', form.validated)
+        finalized = request.data.get('finalized', form.finalized)
+        data = request.data.get('data', None)
+
+        # Update the Form properties and try to save
         try:
-            form = change_form(body)
-        except:
+            form.name = name
+            form.language = language
+            form.country_id = country_id
+            form.ns = ns
+            form.ip_address = ip
+            form.started_at = started_at
+            form.ended_at = ended_at
+            form.submitted_at = submitted_at
+            form.comment = comment
+            form.validated = validated
+            form.finalized = finalized
+            form.save()
+        except Exception:
             return bad_request('Could not change PER form record.')
 
-        if hasattr(form,'status_code') and form.status_code == 400:
```

Comment:
@batpad @thenav56 This felt redundant to me since we're already throwing a `bad_request` above if it fails. Is there any scenario where we might want to keep this in?